Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4699,3 +4705,31 @@ void CGame::HandleCrashDumpEncryption()
     }
 #endif
 }
+
+void CGame::RegisterClientTriggeredEventUsage(CPlayer* pPlayer)
+{
+    if (!pPlayer)
+        return;
+
+    auto iter = m_mapClientTriggeredEvents.find(pPlayer);
+
+    if (iter == m_mapClientTriggeredEvents.end())
+        iter = m_mapClientTriggeredEvents.insert({pPlayer, 0}).first;
+
+    iter->second++;
+}
+
+void CGame::ProcessClientTriggeredEventSpam()
+{
+    for (const auto& pair : m_mapClientTriggeredEvents)
+    {
+        CPlayer* player = pair.first;
```

Comment:
What will happen if a player leaves a server in between of RegisterClientTriggeredEventUsage and ProcessClientTriggeredEventSpam? Formally pointer will be valid but technically it is a segfault.

Reply:
Referencing [CPedSync::UpdateNearPlayersList](https://github.com/multitheftauto/mtasa-blue/blob/abec07aeb90eef19ed47396baed3b190994be54a/Server/mods/deathmatch/logic/CPedSync.cpp#L310-L322), my latest changes should suffice

Reply:
Probably you can crash a server by kicking a player somewhere in between.

Reply:
So, how about moving the call after line 506 `CLOCK_CALL1(m_pACLManager->DoPulse(););`